CommonTools/UtilAlgos/interface/ObjectSelector.h

```diff
@@ -17,6 +17,7 @@
 #include "FWCore/Framework/interface/ConsumesCollector.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/Utilities/interface/InputTag.h"
+#include "CommonTools/CandAlgos/interface/CandDecaySelector.h"'
```

Comment:
Ah, because we do the actual template parameter lookup that chooses to this specialization in ObjectSelector.h at line 36:
```C++
typename StoreManager = typename ::helper::StoreManagerTrait<OutputCollection, edm::EDFilter>::type,
```
Only the parameters that lead clang to the conclusion that this specialization is the best lookup is coming from the mentioned headers. So if we only add the include to those headers, then the next file that ends up getting `reco::CandidateCollection` as the template parameter to ObjectSelector will bring us back to this problem.

So I just added it there to prevent further regressions with this specialization and this class.

Another (even better but less conservative?) solution would be to move the specialization with `reco::CandidateCollection` in the same header where this typedef is. This would prevent this issue in the whole code base.